File: Python/Longest-Common-Prefix.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-
# Longest Common Prefix 53ms


"""
/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015年04月16日10:28:29
 * @url:https://leetcode.com/problems/longest-common-prefix/
 */
"""


class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if not strs:
            return ''
        minLen = len(strs[0])
        index = 0
        sLen = len(strs)
        for i in xrange(1, sLen):
            xLen = len(strs[i])
            if minLen > xLen:
                minLen = xLen
                index = i
        for x in strs:
            k = 0
            while k < minLen:
                if x[k] != strs[index][k]:
                    break
                k += 1
            if k < minLen:
                minLen = k
        return strs[index][:minLen]

File: Java/Longest_Common_Prefix.java

// Longest Common Prefix

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015年04月16日10:28:29
 * @url:https://leetcode.com/problems/longest-common-prefix/
 */

public class Solution {

    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        int len = strs.length;
        int minLen = Integer.MAX_VALUE;
        int index = -1;
        for (int i = 0; i < len; i++) {
            int tempLen = strs[i].length();
            if (minLen > tempLen) {
                minLen = tempLen;
                index = i;
            }
        }
        String shortStr = strs[index];
        for (int i = 0; i < len; i++) {
            int j = 0;
            for (; j < minLen; j++) {
                if (strs[i].charAt(j) != shortStr.charAt(j)) {
                    break;
                }
            }
            if (j < minLen) {
                minLen = j;
            }
        }
        return shortStr.substring(0, minLen);
    }
}

